### PR TITLE
feat: sound groups & instances

### PR DESCRIPTION
**Pull request by IllustrisJack**

### CODE REVIEW

**Comment by HaronK** [comment]
@IllustrisJack can you run `cargo fmt` please?
It would be also good to set "format on save" feature in IDE

File: engine/lib/phx/src/audio/sound.rs [hunk 1]
```diff
@@ -1,32 +1,25 @@
...
 
-use super::process_command_error;
+use super::{process_command_error, SoundInstance};
 
 pub struct Sound {
     path: String,
     sound_data: StaticSoundData,
     emitter: Option<EmitterHandle>,
-    sound_handle: Option<Rc<RefCell<StaticSoundHandle>>>,
+    pub instances: Vec<Rc<RefCell<SoundInstance>>>,
```
Comment (HaronK) [inline engine/lib/phx/src/audio/sound.rs:19]: Better to keep accessibility of all fields the same. Make it private and add getter method.

File: script/GameObjects/Entities/Objects/Projectile.lua [hunk 2]
```diff
@@ -57,7 +57,7 @@ function Entity:addProjectile(source)
 
         --! needs to be handled properly by the audio system eventually
         if volDist > 0 then
-            Config.audio.sounds.pulseFire:Play(volDist)
+            local soundInstance = Config.audio.sounds.pulseFire:Play(volDist)
```
Comment (HaronK) [inline script/GameObjects/Entities/Objects/Projectile.lua:60]: `soundInstance` is not used anywhere

File: engine/lib/phx/src/audio/audio.rs [hunk 3]
```diff
@@ -90,12 +91,19 @@ impl Audio {
         self.audio_manager.num_sounds() as u64
     }
 
-    pub fn get_playing_count(&self) -> u64 {
-        self.sounds
-            .iter()
-            .filter(|sound| sound.borrow().state() == PlaybackState::Playing)
-            .count() as u64
-    }
+    //pub fn get_playing_count(&self) -> u64 {
```
Comment (HaronK) [inline engine/lib/phx/src/audio/audio.rs:94]: Do we still need this? I think it should be on the Lua side now.

**Review by HaronK — APPROVED** [review]

File: engine/lib/phx/src/audio/audio.rs (hunk 3 above)
Comment (IllustrisJack) [reply engine/lib/phx/src/audio/audio.rs:94]: that´s correct :)

**Review by Flatfingers — APPROVED** [review]

**Pull request merged**